CRM/Event/BAO/Participant.php

```diff
@@ -1904,4 +1904,28 @@ public static function getSelfServiceEligibility(int $participantId, string $url
     return $details;
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   * @throws CRM_Core_Exception
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->entity === 'Participant' && $event->action === 'create') {
+      // Set the "registered_by_contact_id" field.
+      // The registered_by_contact_id should always be the person that actually did the registration.
+      // That might be the first participant, but it might be someone registering someone without registering themselves.
+      $event->params['registered_by_contact_id'] = CRM_Core_Session::getLoggedInContactID();
```

Comment:
I think the logic here needs to allow this field to be set. As written, if you call the `Participant::create()` API and pass a value for this field, it will be overwritten and ignored. I think this logic should *only* kick in if the value is omitted.